--- a/chrome/app/chrome_version.rc.version
+++ b/chrome/app/chrome_version.rc.version
@@ -10,7 +10,7 @@
 //
 
 VS_VERSION_INFO VERSIONINFO
- FILEVERSION @MAJOR@,@MINOR@,@BUILD@,@PATCH@
+ FILEVERSION @HELIUM_MAJOR@,@HELIUM_MINOR@,@HELIUM_PATCH@,@HELIUM_PLATFORM@
  PRODUCTVERSION @MAJOR@,@MINOR@,@BUILD@,@PATCH@
  FILEFLAGSMASK 0x17L
 #ifdef _DEBUG
@@ -28,7 +28,7 @@ BEGIN
         BEGIN
             VALUE "CompanyName", "@COMPANY_FULLNAME@"
             VALUE "FileDescription", "@PRODUCT_FULLNAME@"
-            VALUE "FileVersion", "@MAJOR@.@MINOR@.@BUILD@.@PATCH@"
+            VALUE "FileVersion", "@HELIUM_MAJOR@.@HELIUM_MINOR@.@HELIUM_PATCH@.@HELIUM_PLATFORM@"
             VALUE "InternalName", "@INTERNAL_NAME@"
             VALUE "LegalCopyright", "@COPYRIGHT@"
             VALUE "OriginalFilename", "@ORIGINAL_FILENAME@"
--- a/chrome/app/main_dll_loader_win.cc
+++ b/chrome/app/main_dll_loader_win.cc
@@ -150,7 +150,7 @@ base::FilePath GetModulePath(std::wstrin
   // executable's directory and return the path if it can be read. This is the
   // expected location of modules for proper installs.
   const base::FilePath module_path =
-      exe_dir.AppendASCII(chrome::kChromeVersion).Append(module_name);
+      exe_dir.AppendASCII(chrome::kHeliumVersion).Append(module_name);
   if (ModuleCanBeRead(module_path))
     return module_path;
 
--- a/chrome/app/version_assembly/BUILD.gn
+++ b/chrome/app/version_assembly/BUILD.gn
@@ -40,6 +40,6 @@ windows_manifest("chrome_exe_manifest")
 process_version("version_assembly_manifest") {
   template_file = "version_assembly_manifest.template"
   sources = [ "//chrome/VERSION" ]
-  output = "$root_build_dir/$chrome_version_full.manifest"
+  output = "$root_build_dir/$helium_version_full.manifest"
   process_only = true
 }
--- a/chrome/app/version_assembly/chrome_exe_manifest.template
+++ b/chrome/app/version_assembly/chrome_exe_manifest.template
@@ -3,8 +3,8 @@
   <dependency>
   <dependentAssembly>
     <assemblyIdentity type='win32'
-      name='@MAJOR@.@MINOR@.@BUILD@.@PATCH@'
-      version='@MAJOR@.@MINOR@.@BUILD@.@PATCH@' language='*'/>
+      name='@HELIUM_MAJOR@.@HELIUM_MINOR@.@HELIUM_PATCH@.@HELIUM_PLATFORM@'
+      version='@HELIUM_MAJOR@.@HELIUM_MINOR@.@HELIUM_PATCH@.@HELIUM_PLATFORM@' language='*'/>
   </dependentAssembly>
   </dependency>
 </assembly>
\ No newline at end of file
--- a/chrome/app/version_assembly/version_assembly_manifest.template
+++ b/chrome/app/version_assembly/version_assembly_manifest.template
@@ -1,8 +1,8 @@
 <assembly
   xmlns='urn:schemas-microsoft-com:asm.v1' manifestVersion='1.0'>
   <assemblyIdentity
-      name='@MAJOR@.@MINOR@.@BUILD@.@PATCH@'
-      version='@MAJOR@.@MINOR@.@BUILD@.@PATCH@'
+      name='@HELIUM_MAJOR@.@HELIUM_MINOR@.@HELIUM_PATCH@.@HELIUM_PLATFORM@'
+      version='@HELIUM_MAJOR@.@HELIUM_MINOR@.@HELIUM_PATCH@.@HELIUM_PLATFORM@'
       type='win32'/>
   <file name='chrome_elf.dll'/>
 </assembly>
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -1338,7 +1338,7 @@ base::FilePath GetModulePath(std::wstrin
   // executable's directory and return the path if it can be read. This is the
   // expected location of modules for proper installs.
   const base::FilePath module_path =
-      exe_dir.AppendASCII(chrome::kChromeVersion).Append(module_name);
+      exe_dir.AppendASCII(chrome::kHeliumVersion).Append(module_name);
   if (base::PathExists(module_path)) {
     return module_path;
   }
--- a/chrome/browser/web_applications/chrome_pwa_launcher/chrome_pwa_launcher_util.cc
+++ b/chrome/browser/web_applications/chrome_pwa_launcher/chrome_pwa_launcher_util.cc
@@ -21,7 +21,7 @@ base::FilePath GetChromePwaLauncherPath(
   base::FilePath chrome_dir;
   if (!base::PathService::Get(base::DIR_EXE, &chrome_dir))
     return base::FilePath();
-  base::FilePath launcher_path = chrome_dir.AppendASCII(chrome::kChromeVersion)
+  base::FilePath launcher_path = chrome_dir.AppendASCII(chrome::kHeliumVersion)
                                      .Append(kChromePwaLauncherExecutable);
   if (base::PathExists(launcher_path))
     return launcher_path;
--- a/chrome/common/chrome_constants.cc
+++ b/chrome/common/chrome_constants.cc
@@ -12,6 +12,7 @@
 namespace chrome {
 
 const char kChromeVersion[] = CHROME_VERSION_STRING;
+const char kHeliumVersion[] = HELIUM_VERSION_STRING;
 
 // The following should not be used for UI strings; they are meant
 // for system strings only. UI changes should be made in the GRD.
--- a/chrome/common/chrome_constants.h
+++ b/chrome/common/chrome_constants.h
@@ -15,6 +15,7 @@
 namespace chrome {
 
 extern const char kChromeVersion[];
+extern const char kHeliumVersion[];
 extern const base::FilePath::CharType kBrowserProcessExecutableName[];
 extern const base::FilePath::CharType kHelperProcessExecutableName[];
 extern const base::FilePath::CharType kBrowserProcessExecutablePath[];
--- a/chrome/common/chrome_version.h.in
+++ b/chrome/common/chrome_version.h.in
@@ -14,6 +14,9 @@
 #define CHROME_VERSION_BUILD @BUILD@
 #define CHROME_VERSION_PATCH @PATCH@
 
+#define HELIUM_VERSION_STRING \
+  "@HELIUM_MAJOR@.@HELIUM_MINOR@.@HELIUM_PATCH@.@HELIUM_PLATFORM@"
+
 // Branding Information
 
 #define COMPANY_FULLNAME_STRING "@COMPANY_FULLNAME@"
--- a/chrome/installer/mini_installer/mini_installer_exe_version.rc.version
+++ b/chrome/installer/mini_installer/mini_installer_exe_version.rc.version
@@ -8,7 +8,7 @@
 // SDK versions which causes headaches building in some environments. The
 // VERSIONINFO resource will always be at index 1.
 1 VERSIONINFO
- FILEVERSION @MAJOR@,@MINOR@,@BUILD@,@PATCH@
+ FILEVERSION @HELIUM_MAJOR@,@HELIUM_MINOR@,@HELIUM_PATCH@,@HELIUM_PLATFORM@
  PRODUCTVERSION @MAJOR@,@MINOR@,@BUILD@,@PATCH@
  FILEFLAGSMASK 0x17L
 #ifdef _DEBUG
@@ -26,7 +26,7 @@ BEGIN
         BEGIN
             VALUE "CompanyName", "@COMPANY_FULLNAME@"
             VALUE "FileDescription", "@PRODUCT_INSTALLER_FULLNAME@"
-            VALUE "FileVersion", "@MAJOR@.@MINOR@.@BUILD@.@PATCH@"
+            VALUE "FileVersion", "@HELIUM_MAJOR@.@HELIUM_MINOR@.@HELIUM_PATCH@.@HELIUM_PLATFORM@"
             VALUE "InternalName", "mini_installer"
             VALUE "LegalCopyright", "@COPYRIGHT@"
             VALUE "ProductName", "@PRODUCT_INSTALLER_FULLNAME@"
--- a/chrome/installer/setup/BUILD.gn
+++ b/chrome/installer/setup/BUILD.gn
@@ -114,6 +114,7 @@ if (is_win) {
       "//components/crash/core/common",
       "//components/metrics:client_info",
       "//components/metrics:metrics_pref_names",
+      "//base/version_info",
       "//components/version_info:channel",
       "//third_party/crashpad/crashpad/client",
       "//third_party/crashpad/crashpad/util",
--- a/chrome/installer/setup/install_worker.cc
+++ b/chrome/installer/setup/install_worker.cc
@@ -34,6 +34,7 @@
 #include "base/strings/utf_string_conversions.h"
 #include "base/version.h"
 #include "base/version_info/channel.h"
+#include "base/version_info/version_info.h"
 #include "base/win/registry.h"
 #include "base/win/security_util.h"
 #include "base/win/sid.h"
@@ -636,10 +637,11 @@ void AddUninstallShortcutWorkItems(const
                                          InstallUtil::GetPublisherName(), true);
     install_list->AddSetRegValueWorkItem(
         reg_root, uninstall_reg, KEY_WOW64_32KEY, L"Version",
-        ASCIIToWide(new_version.GetString()), true);
+        ASCIIToWide(version_info::GetHeliumVersionNumber()), true);
     install_list->AddSetRegValueWorkItem(
         reg_root, uninstall_reg, KEY_WOW64_32KEY, L"DisplayVersion",
-        ASCIIToWide(new_version.GetString()), true);
+        ASCIIToWide(version_info::GetHeliumVersionNumber()),
+        true);
     // TODO(wfh): Ensure that this value is preserved in the 64-bit hive when
     // 64-bit installs place the uninstall information into the 64-bit registry.
     install_list->AddSetRegValueWorkItem(reg_root, uninstall_reg,
--- a/chrome/installer/setup/setup_main.cc
+++ b/chrome/installer/setup/setup_main.cc
@@ -587,7 +587,7 @@ installer::InstallStatus RenameChromeExe
                                     WorkItem::MoveTreeOptions{});
 
   AddFinalizeUpdateWorkItems(original_state,
-                             base::Version(chrome::kChromeVersion),
+                             base::Version(chrome::kHeliumVersion),
                              *installer_state, setup_exe, install_list.get());
 
   // Add work items to delete Chrome's "opv", "cpv", and "cmd" values.
--- a/chrome/tools/build/win/create_installer_archive.py
+++ b/chrome/tools/build/win/create_installer_archive.py
@@ -47,25 +47,22 @@
     """Returns the full build version string constructed from information in
     VERSION_FILE.  Any segment not found in that file will default to '0'.
     """
-    major = 0
-    minor = 0
-    build = 0
-    patch = 0
+    values = {}
     # The version file is located at the directory ${CHROMIUM_SRC}/chrome.
     version_file_path = os.path.join(
         os.path.dirname(__file__), '../../..', VERSION_FILE
     )
     for line in open(version_file_path, 'r'):
-        line = line.rstrip()
-        if line.startswith('MAJOR='):
-            major = line[6:]
-        elif line.startswith('MINOR='):
-            minor = line[6:]
-        elif line.startswith('BUILD='):
-            build = line[6:]
-        elif line.startswith('PATCH='):
-            patch = line[6:]
-    return '%s.%s.%s.%s' % (major, minor, build, patch)
+        key, sep, value = line.rstrip().partition('=')
+        if sep:
+            values[key] = value
+    return '%s.%s.%s.%s' % (
+        values.get('HELIUM_MAJOR', '0'),
+        values.get('HELIUM_MINOR', '0'),
+        values.get('HELIUM_PATCH', '0'),
+        values.get('HELIUM_PLATFORM', '0'),
+    )
+
 
 
 def CompressUsingLZMA(
--- a/chrome/version.gni
+++ b/chrome/version.gni
@@ -19,7 +19,9 @@
 # all values we need at once.
 _version_dictionary_template = "full = \"@MAJOR@.@MINOR@.@BUILD@.@PATCH@\" " +
                                "major = \"@MAJOR@\" minor = \"@MINOR@\" " +
-                               "build = \"@BUILD@\" patch = \"@PATCH@\" "
+                               "build = \"@BUILD@\" patch = \"@PATCH@\" " +
+                               "helium_full = \"@HELIUM_MAJOR@.@HELIUM_MINOR@" +
+                               ".@HELIUM_PATCH@.@HELIUM_PLATFORM@\" "
 
 # The file containing the Chrome version number.
 chrome_version_file = "//chrome/VERSION"
@@ -161,6 +163,7 @@ _result = exec_script("//build/util/vers
 
 # Full version. For example "45.0.12321.0"
 chrome_version_full = _result.full
+helium_version_full = _result.helium_full
 
 # The constituent parts of the full version.
 chrome_version_major = _result.major
--- a/chrome/installer/setup/last_breaking_installer_version.cc
+++ b/chrome/installer/setup/last_breaking_installer_version.cc
@@ -11,12 +11,8 @@ namespace installer {
 // When updating the documentation, keep a history of maximum 3 milestones prior
 // the current version because downgrades are supported for 3 milestones.
 //
-// Breaking changes from 85.0.4169.0:
-//  Change: Default installation directory for fresh 64 bits browsers moved from
-//    base::DIR_PROGRAM_FILESX86 to DIR_PROGRAM_FILES.
-//  Reason for being breaking: Downgrading to previous version will result in
-//    stale data in DIR_PROGRAM_FILES since the previous installers do not know
-//    if there is Chrome data at this location.
-const wchar_t kLastBreakingInstallerVersion[] = L"85.0.4169.0";
+// Helium uses its own version scheme, which is below every Chromium
+// "breaking installer version".
+const wchar_t kLastBreakingInstallerVersion[] = L"0.0.0.0";
 
 }  // namespace installer
